consensus/types/src/sync_subnet_id.rs

```diff
@@ -33,6 +36,20 @@ impl SyncSubnetId {
     pub fn new(id: u64) -> Self {
         id.into()
     }
+
+    /// Compute required subnets to subscribe to given the sync committee indices.
+    pub fn compute_subnets_for_sync_committee<T: EthSpec>(
+        sync_committee_indices: Vec<u64>,
+        spec: &ChainSpec,
+    ) -> Result<Vec<Self>, ArithError> {
+        let subnet_size =
+            T::SyncCommitteeSize::to_u64().safe_div(spec.sync_committee_subnet_count)?;
+
+        sync_committee_indices
+            .into_iter()
+            .map(|index| index.safe_div(subnet_size).map(Self::new))
+            .collect::<Result<_, _>>()
```

Comment:
The spec [uses a set](https://github.com/ethereum/eth2.0-specs/blob/v1.1.0-alpha.7/specs/altair/validator.md#broadcast-sync-committee-message) for this return type, to account for the possibility of a validator being assigned multiple times to the same subnet. I implemented the same function for the VC, but will switch to using your implementation. I used `dedup` to get the set semantics (assuming `sync_committee_indices` is sorted).

```rust
    pub fn subnet_ids<E: EthSpec>(&self) -> Result<Vec<u64>, ArithError> {
        let mut subnets = self
            .validator_sync_committee_indices
            .iter()
            .map(|position| {
                position
                    .safe_div(E::SyncCommitteeSize::to_u64().safe_div(SYNC_COMMITTEE_SUBNET_COUNT)?)
            })
            .collect::<Result<Vec<_>, _>>()?;
        subnets.dedup();
        Ok(subnets)
    }
```

@realbigsean also has a version of this method that returns `HashMap<SyncSubnetId, Vec<usize>>`, but I'm happy to keep that separate (https://github.com/sigp/lighthouse/pull/2321#discussion_r647163764)

Reply:
We could also just return a `HashSet` from the function instead of a `Vec` right? Wouldn't have to make assumptions about `sync_committee_indices` being sorted then.

Reply:
Yeah actually that's a lot safer